packages/ra-ui-materialui/src/list/SimpleList.tsx

```diff
@@ -95,15 +96,18 @@ const SimpleList: FC<SimpleListProps> = props => {
     return (
         total > 0 && (
             <List className={className} {...sanitizeListRestProps(rest)}>
-                {ids.map(id => (
+                {ids.map((id, rowIndex) => (
                     <LinkOrNot
                         linkType={linkType}
                         basePath={basePath}
                         id={id}
                         key={id}
                         record={data[id]}
                     >
-                        <ListItem button={!!linkType as any}>
+                        <ListItem
+                            button={!!linkType as any}
+                            style={rowStyle ? rowStyle(data[id], rowIndex) : null}
```

Comment:
I'd prefer using `undefined` instead of `null` in this case.

Reply:
Sure, will change that!